src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -69,7 +69,12 @@ private async Task RenderComponentCore(HttpContext context)
             handler: result.HandlerName,
             form: result.HandlerName != null && context.Request.HasFormContentType ? await context.Request.ReadFormAsync() : null);
 
-        await using var writer = CreateResponseWriter(context.Response.Body);
+        // Matches MVC's MemoryPoolHttpResponseStreamWriterFactory.DefaultBufferSize
+        var defaultBufferSize = 16 * 1024;
+        await using var writer = new HttpResponseStreamWriter(context.Response.Body, Encoding.UTF8, defaultBufferSize, ArrayPool<byte>.Shared, ArrayPool<char>.Shared);
+        var viewBufferScope = new MemoryPoolViewBufferScope(ArrayPool<ViewBufferValue>.Shared);
+        var viewBuffer = new MemoryPoolViewBuffer(viewBufferScope, pageComponent.FullName ?? pageComponent.Name, defaultBufferSize);
+        await using var bufferWriter = new ViewBufferTextWriter(viewBuffer, Encoding.UTF8, HtmlEncoder.Default, writer);
```

Comment:
We should have fixed #47477 while doing this change. It should be as simple as grabbing `HtmlEncoder` from DI and using that instead of `HtmlEncoder.Default` (and ensuring `IServiceCollection.AddWebEncoders()` is called by `IServiceCollection.AddRazorComponents()`)